lib/seccomp/unittest/CSystemCallFilterTest.cc

```diff
@@ -0,0 +1,216 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+#include "CSystemCallFilterTest.h"
+
+#include <core/CLogger.h>
+#include <core/COsFileFuncs.h>
+#include <core/CSleep.h>
+#include <core/CThread.h>
+#ifdef Linux
+#include <core/CRegex.h>
+#include <core/CUname.h>
+#endif
+
+#include <seccomp/CSystemCallFilter.h>
+
+#include <test/CTestTmpDir.h>
+
+#include <cstdlib>
+#include <string>
+
+namespace {
+
+const uint32_t SLEEP_TIME_MS = 100;
+const size_t TEST_SIZE = 10000;
+const size_t MAX_ATTEMPTS = 20;
+const char TEST_CHAR = 'a';
+#ifdef Windows
+const char* TEST_READ_PIPE_NAME = "\\\\.\\pipe\\testreadpipe";
+const char* TEST_WRITE_PIPE_NAME = "\\\\.\\pipe\\testwritepipe";
+#else
+const char* TEST_READ_PIPE_NAME = "testreadpipe";
+const char* TEST_WRITE_PIPE_NAME = "testwritepipe";
+#endif
+
+
+class CNamedPipeWriter : public ml::core::CThread {
+public:
+    CNamedPipeWriter(const std::string& fileName, size_t size)
+        : m_FileName(fileName), m_Size(size) {}
+protected:
+    virtual void run() {
+        // Wait for the file to exist
+        ml::core::CSleep::sleep(SLEEP_TIME_MS);
+
+        std::ofstream strm(m_FileName.c_str());
+        for (size_t i = 0; i < m_Size && strm.good(); ++i) {
+            strm << TEST_CHAR;
+        }
+    }
+
+    virtual void shutdown() {}
+
+private:
+    std::string m_FileName;
+    size_t m_Size;
+};
+
+class CNamedPipeReader : public ml::core::CThread {
+public:
+    CNamedPipeReader(const std::string& fileName) : m_FileName(fileName) {}
+
+    const std::string& data() const { return m_Data; }
+
+protected:
+    virtual void run() {
+        m_Data.clear();
+
+        std::ifstream strm;
+
+        // Try to open the file repeatedly to allow time for the other
+        // thread to create it
+        size_t attempt(1);
+        do {
+            CPPUNIT_ASSERT(attempt++ <= MAX_ATTEMPTS);
+            ml::core::CSleep::sleep(SLEEP_TIME_MS);
+            strm.open(m_FileName.c_str());
+        } while (!strm.is_open());
+
+        static const std::streamsize BUF_SIZE = 512;
+        char buffer[BUF_SIZE];
+        while (strm.good()) {
+            strm.read(buffer, BUF_SIZE);
+            CPPUNIT_ASSERT(!strm.bad());
+            if (strm.gcount() > 0) {
+                // This code deals with the test character we write to
+                // detect the short-lived connection problem on Windows
+                const char* copyFrom = buffer;
+                size_t copyLen = static_cast<size_t>(strm.gcount());
+                if (m_Data.empty() && *buffer == ml::core::CNamedPipeFactory::TEST_CHAR) {
+                    ++copyFrom;
+                    --copyLen;
+                }
+                if (copyLen > 0) {
+                    m_Data.append(copyFrom, copyLen);
+                }
+            }
+        }
+    }
+
+    virtual void shutdown() {}
+
+private:
+    std::string m_FileName;
+    std::string m_Data;
+};
+}
```

Comment:
The end of the unnamed namespace could be moved below the `systemCall()` function below.